extract/ext/pydecompile.py

```diff
@@ -1,11 +1,35 @@
+# Monkeypatch uncompyle6's print_doctstring function.
+# Unfortunately we can't get in early enough due to way __init__ is structured, so each use case must be patched.
+import uncompyle6
+
+orig_print_docstring = uncompyle6.pysource.print_docstring
+
+
+def patched_print_docstring(self, indent, docstring):
+    """Function to monkey patch the handling of binary docstrings."""
+    if isinstance(docstring, bytes):
+        docstring = docstring.decode("utf8", errors="replace")
```

Comment:
I think we may be able to use `backslashreplace` instead of `replace`, as it will preserve the bytes from being lost. The print_docstring method will add an `r` in front of the docstring, which may not be exactly the same, but closer.

Reply:
Interesting. I forgot `backslashreplace` was an option!
While the result ending up as a raw string isn't a 100% reproduction of the original source, I agree it's a much better way to at least make the escapes viewable in a human readable way. (and as you found, byte strings get dropped when python complies its own source in higher python versions :)